planner/core/show_predicate_extractor.go

```diff
@@ -58,8 +58,12 @@ func (e *ShowColumnsTableExtractor) Extract(show *ast.ShowStmt) bool {
 		switch pattern.Pattern.(type) {
 		case *driver.ValueExpr:
 			// It is used in `SHOW COLUMNS FROM t LIKE `abc``.
+			escape := "?"
+			if _, ok := pattern.Escape.(*driver.ParamMarkerExpr); !ok {
+				escape = pattern.Escape.(*driver.ValueExpr).GetString()
+			}
```

Comment:
`SHOW COLUMNS FROM t LIKE abc` cannot use escape, see the parser. I think it will report a syntax error

Reply:
Hi, this is right from my test result, "show columns from t like abc" can run correctly, and also "show tables like ...” also run correctly

Reply:
You are right, the prepared case for "show columns ...." syntax error. let me check more, THKS much

Reply:
Yes, you are right! "prepared stmt1 from 'show columns from t like abc escape ?'" and single "show columns from t like abc escape '#'" both syntax error. What I'm doubt is: this issue is about prepared stmt not compatible with regular SQL syntax, that is: select * from t1 where a like '...' escape '.' works, but prepared not works. Now the "show columns.." and "show tables..." both not works. so do I need to fix these in this issue, or should in another issue?

Reply:
I added logic in parser.y to make "show columns" "show tables" syntax work for escape and prepared cases. please check

Reply:
no, keep the original logic for `show tables` and `show columns`, please follow the MySQL syntax here.